framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1180,6 +1188,7 @@ public AnnotatedTypeMirror fromElement(Element elt) {
                 && (typesFromStubFiles == null || !typesFromStubFiles.containsKey(elt))) {
             type = toAnnotatedType(elt.asType(), ElementUtils.isTypeDeclaration(elt));
             ElementAnnotationApplier.apply(type, elt, this);
+            typeVarAnnotator.visit(type);
```

Comment:
Both places that use the `typeVarAnnotator` call `ElementAnnotationApplier.apply(type, elt, this);` right before. Conceptually, shouldn't this action already happen there?
Did you look why the `TypeVarUseApplier` doesn't already do this or whether you can move this additional logic there?

Reply:
I've done this.